storage/perfschema/pfs_variable.cc

```diff
@@ -1376,6 +1389,16 @@ void PFS_status_variable_cache::manifest(THD *thd,
       SHOW_FUNC resolves to another SHOW_FUNC.
     */
     if (show_var_ptr->type == SHOW_FUNC) {
+      if (before_wild_len > 0) {
+        const auto len = strlen(show_var_ptr->name);
+        const auto min_len = std::min(before_wild_len, len);
+        if (!my_strnncoll(system_charset_info,
```

Comment:
I'm seeing a compiler issue here:

```
/home/herman/rocks-mysql/8.0/storage/perfschema/pfs_variable.cc:1395:13: error: logical not is only applied to the left hand side of this comparison
      [-Werror,-Wlogical-not-parentheses]
        if (!my_strnncoll(system_charset_info,
            ^
/home/herman/rocks-mysql/8.0/storage/perfschema/pfs_variable.cc:1395:13: note: add parentheses after the '!' to evaluate the comparison first
        if (!my_strnncoll(system_charset_info,
            ^
             (
/home/herman/rocks-mysql/8.0/storage/perfschema/pfs_variable.cc:1395:13: note: add parentheses around left hand side expression to silence this warning
        if (!my_strnncoll(system_charset_info,
            ^
            (
```

While the code seems correct as written "!my_strnncoll() == 0", I think it might be clearer as my_strnncoll() != 0?

Reply:
It's fixed